src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -88,6 +114,8 @@ public CourseScoresDTO calculateCourseScores(long courseId, Collection<Long> stu
         }
         else {
             var participations = studentParticipationRepository.findByCourseIdWithRelevantResult(courseId);
+            // Only use those participations where one of the students handed to this method actually takes part.
+            // For the single student case, this is done in the db query.
             var studentIdSet = new HashSet<>(studentIds);
             for (StudentParticipation participation : participations) {
                 for (User student : participation.getStudents()) {
```

Comment:
Why do we need that additional conversion from `studentIds` to `studentIdSet`?

Reply:
`studentIds` is a `List` at this point. I guess the conversion to a `Set` happens for performance reasons as the `contains` method runs much quicker for sets.